apis/hive/v1/agentclusterinstall_types.go

```diff
@@ -0,0 +1,155 @@
+// +kubebuilder:validation:Enum="";Enabled;Disabled
+type HyperthreadingMode string
+
+const (
+	// HyperthreadingEnabled indicates that hyperthreading is enabled.
+	HyperthreadingEnabled HyperthreadingMode = "Enabled"
+	// HyperthreadingDisabled indicates that hyperthreading is disabled.
+	HyperthreadingDisabled HyperthreadingMode = "Disabled"
+)
+
+// AgentClusterInstallSpec defines the desired state of the AgentClusterInstall.
+type AgentClusterInstallSpec struct {
+
+	// ImageSetRef is a reference to a ClusterImageSet. The release image specified in the ClusterImageSet will be used
+	// to install the cluster.
+	ImageSetRef ClusterImageSetReference `json:"imageSetRef"`
+
+	// ClusterDeploymentRef is a reference to the ClusterDeployment associated with this AgentClusterInstall.
+	ClusterDeploymentRef corev1.LocalObjectReference `json:"clusterDeploymentRef"`
+
+	// ClusterMetadata contains metadata information about the installed cluster. It should be populated once the cluster install is completed. (it can be populated sooner if desired, but Hive will not copy back to ClusterDeployment until the Installed condition goes True.
+	ClusterMetadata *ClusterMetadata `json:"clusterMetadata,omitempty"`
+
+	// ManifestsConfigMapRef is a reference to user-provided manifests to
+	// add to or replace manifests that are generated by the installer.
+	ManifestsConfigMapRef *corev1.LocalObjectReference `json:"manifestsConfigMapRef,omitempty"`
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	Networking Networking `json:"networking"`
+
+	// SSHPublicKey will be added to all cluster hosts for use in debugging.
+	// +optional
+	SSHPublicKey string `json:"sshPublicKey,omitempty"`
+
+	// ProvisionRequirements defines configuration for when the installation is ready to be launched automatically.
+	ProvisionRequirements ProvisionRequirements `json:"provisionRequirements"`
+
+	// ControlPlane is the configuration for the machines that comprise the
+	// control plane.
+	// +optional
+	ControlPlane *AgentMachinePool `json:"controlPlane,omitempty"`
+
+	// Compute is the configuration for the machines that comprise the
+	// compute nodes.
+	// +optional
+	Compute []AgentMachinePool `json:"compute,omitempty"`
+}
+
+// ProvisionRequirements defines configuration for when the installation is ready to be launched automatically.
+type ProvisionRequirements struct {
+
+	// ControlPlaneAgents is the number of matching approved and ready Agents with the control plane role
+	// required to launch the install. Must be either 1 or 3.
+	ControlPlaneAgents int `json:"controlPlaneAgents"`
+
+	// WorkerAgents is the minimum number of matching approved and ready Agents with the worker role
+	// required to launch the install.
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	WorkerAgents int `json:"workerAgents,omitempty"`
+}
+
+// Networking defines the pod network provider in the cluster.
```

Comment:
there are already defined in hive/v1/agent/installstrategy.go ?? we should try and not crowd the hive/v1 package with these specific types..

Reply:
hive/v1/agent/installstrategy.go will die off, I discussed with assisted team that we won't yank the API until we complete transition, but this is meant to fully obsolete that.

Reply:
Actually given we're in a branch, we can probably remove the old. I'm going to go do that now, I also have to move some missed InstallStrategyStatus over.

Reply:
I still don't think we should add a Networking struct to hive/v1 when we currently know this is only meant to be used with agent cluster install. I would recommend we move this to hive/v1/agent so that this Networking doesn't become our fo to for all the other types we might add later on.

Reply:
I actually tried that at first but it caused a circular dependency. Now that we've killed off the old types though it might be better. We ref from ClusterDeployment.Spec.Platform into agent package, I think the only remaining problem is AgentClusterInstall.Spec.ImageSet ref being hivev1.ClusterImageSetReference. We could duplicate it, wdyt?

Reply:
i don't understand what caused the circular dependency? hive/v1/agent has no import to hive/v1 ...

Reply:
Abhinav clarified in scrum that this was referring just to the sub-structs like Networking and others, to keep them out of the main hivev1 package and from being used inadvertently. I will relocate those.